src/core/connection.c

```diff
@@ -6567,6 +6660,217 @@ QuicConnParamSet(
         return QUIC_STATUS_SUCCESS;
     }
 
+    case QUIC_PARAM_CONN_ADD_LOCAL_ADDRESS: {
```

Comment:
The code for handling all this is quite complex, and I think it deserves to be put in a separate function.